### ISSUE TITLE
Feature: Elliptic Curve support

### ISSUE THREAD

**Issue by robotdan** [FusionAuth/fusionauth-jwt#4]
### Overview

The JSON Web Algorithms (JWA) RFC (https://tools.ietf.org/html/rfc7518) specifies several cryptography standards, one of the optional is Elliptic Curve support. 

In section 3.1.  "alg" (Algorithm) Header Parameter Values for JWS - the following algorithms are recommended for implementation.

```
   | ES256        | ECDSA using P-256 and SHA-256  | Recommended+       |
   | ES384        | ECDSA using P-384 and SHA-384  | Optional           |
   | ES512        | ECDSA using P-521 and SHA-512  | Optional           |
```

### 

Add support for these algorithms by adding a new `Verifier` and `Signer` impl in a new library called `fusionauth-jwt-elliptic`. 

### Example Verifier Impl Stub

```java
public class EllipticVerifier implements Verifier {

  private final byte[] secret;

  private EllipticVerifier(String secret) {
    Objects.requireNonNull(secret);
    this.secret = secret.getBytes(StandardCharsets.UTF_8);
  }

  public static EllipticVerifier newVerifier(String secret) {
    return new EllipticVerifier(secret);
  }

  @Override
  public boolean canVerify(Algorithm algorithm) {
    switch (algorithm) {
      case ES256:
      case ES384:
      case ES512:
        return true;
      default:
        return false;
    }
  }

  @Override
  public void verify(Algorithm algorithm, byte[] payload, byte[] signature) {
    Objects.requireNonNull(algorithm);
    Objects.requireNonNull(payload);
    Objects.requireNonNull(signature);

    // Make me work!
  }
}
```

### Example Signer Impl Stub

```java
public class EllipticSigner implements Signer {

  private final Algorithm algorithm;

  private byte[] secret;

  private Elliptic(Algorithm algorithm, String secret) {
    this.algorithm = algorithm;
    this.secret = secret.getBytes(StandardCharsets.UTF_8);
  }

  public static EllipticSigner newSHA256Signer(String secret) {
    return new HMACSigner(Algorithm.ES256, secret);
  }

  public static EllipticSigner newSHA384Signer(String secret) {
    return new EllipticSigner(Algorithm.ES384, secret);
  }

  public static EllipticSigner newSHA512Signer(String secret) {
    return new EllipticSigner(Algorithm.ES512, secret);
  }

  @Override
  public Algorithm getAlgorithm() {
    return algorithm;
  }

  @Override
  public byte[] sign(String message) {
    Objects.requireNonNull(algorithm);
    Objects.requireNonNull(secret);

    // Make me work!
  }
}
```

**Comment by robotdan** [comment]
In progress

**Comment by robotdan** [comment]
Available in version 3.0.0
